ansible_base/rbac/models.py

```diff
@@ -44,7 +44,38 @@ def __str__(self):
         return f"<{self.__class__.__name__}: {self.codename}>"
 
 
+class ManagedRoleFromSetting:
+    def __init__(self, role_name):
+        super().__init__()
+        self.role_name = role_name
+
+    def __get__(self, obj, objtype=None):
+        if obj is None:
+            return self
+        if self.role_name not in obj._cache:
+            obj._cache[self.role_name] = RoleDefinition.objects.filter(name=self.role_name).first()
+        return obj._cache[self.role_name]
+
+
+class ManagedRoleManager:
+    def __init__(self):
+        self._cache = {}
+
+    def clear(self) -> None:
+        "Clear any managed roles already loaded into the cache"
+        self._cache = {}
+
+    org_admin = ManagedRoleFromSetting('Organization Admin')
+    org_member = ManagedRoleFromSetting('Organization Member')
+    team_admin = ManagedRoleFromSetting('Team Admin')
+    team_member = ManagedRoleFromSetting('Team Member')
```

Comment:
Would we put System Auditor in here as well or, because not all services support it, would we have a service extend this class to add it? If a service wanted to extend this what would that look like since its hardcoded on line 77 below?

Reply:
"service wanted to extend this" - I'm not doing that here. My approach:

 - introduce a caching data structure for managed roles
 - introduce a registry, a part of that, including the algorithm to create the managed role

so this is doing the first part but not the second part. They will interact, but we'll have the caching in any case.